pyrmq/tests/test_consumer.py

```diff
@@ -116,3 +121,36 @@ def error_callback(*args):
 
             wait_for_result(response, [1])
             consumer.close()
+
+
+def should_get_message_with_higher_priority(priority_session: Publisher):
+    for i in range(0, 10):
+        body = {"test": f"test{i}"}
+        if i >= TEST_ARGUMENTS["x-max-priority"]:
+            first_expected = body
+        priority_session.publish(body, priority=i)
+    response = {}
+
+    last_expected = {"test": "test0"}
+
+    def callback(data):
+        if not response:
+            """
+            Checks the first message if it has the highest priority.
+            Dependent on ``x-max-priority`` argument. Test max priority is 5
+            Messages with priority 5 and above will be treated as priority 5.
+            Thus, they will be queued after the first message received with the highest priority
+            """
+            assert first_expected == data
+        response.update(data)
+
+    consumer = Consumer(
+        exchange_name=priority_session.exchange_name,
+        queue_name=priority_session.queue_name,
+        routing_key=priority_session.routing_key,
+        callback=callback,
+    )
+    consumer.start()
+    # Last message received would always be message with lowest priority
```

Comment:
This is not always true. If all messages have a priority higher than max priority, then the last that will be consumed is the last one that was published.